src/data/matrix/basic.lean

```diff
@@ -257,22 +295,48 @@ by simp_rw [dot_product, mul_comm]
   dot_product v w = v ⟨⟩ * w ⟨⟩ :=
 by simp [dot_product]
 
-@[simp] lemma dot_product_zero [non_unital_non_assoc_semiring α] (v : m → α) :
-  dot_product v 0 = 0 :=
+section dot_product_zero
+
+variables [non_unital_non_assoc_semiring α] (v : m → α)
+
+@[simp] lemma dot_product_zero  : dot_product v 0 = 0 :=
 by simp [dot_product]
 
-@[simp] lemma dot_product_zero' [non_unital_non_assoc_semiring α] (v : m → α) :
-  dot_product v (λ _, 0) = 0 :=
+@[simp] lemma dot_product_zero'  : dot_product v (λ _, 0) = 0 :=
 dot_product_zero v
 
-@[simp] lemma zero_dot_product [non_unital_non_assoc_semiring α] (v : m → α) :
-  dot_product 0 v = 0 :=
+@[simp] lemma zero_dot_product  : dot_product 0 v = 0 :=
 by simp [dot_product]
 
-@[simp] lemma zero_dot_product' [non_unital_non_assoc_semiring α] (v : m → α) :
-  dot_product (λ _, (0 : α)) v = 0 :=
+@[simp] lemma zero_dot_product' : dot_product (λ _, (0 : α)) v = 0 :=
 zero_dot_product v
 
+end dot_product_zero
```

Comment:
I extracted this tidy-up to #8325 to try and reduce the size of this PR - I've made sure to keep you as a co-author.

Reply:
Sure, Eric. Is there anything I need to do now for this PR?

Reply:
Nope, just letting you know.